### PR TITLE
Added hardware event schema for Redfish events

### PR DESCRIPTION
**Pull request by jzding**

### CODE REVIEW

File: pkg/channel/data.go [hunk 1]
```diff
@@ -31,5 +30,5 @@ type DataChan struct {
 	// if not set then the data is sent to out channel and processed by side car  default method
 	OnReceiveOverrideFn func(e cloudevents.Event) error
 	// ProcessEventFn  Optional, this allows to customize message handler thar was received at the out channel
-	ProcessEventFn func(e event.Event) error
+	ProcessEventFn func(e interface{}) error
```
Comment (aneeshkp) [inline pkg/channel/data.go:33]: will this slow down, by checking for type on every event we get ?
@atyronesmith what are your thoughts? trying to accommodate both hardware schema and cloud-native schema

File: pkg/hwevent/event_ce.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"encoding/json"
+	"fmt"
+
+	cloudevent "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+)
+
+//NewCloudEvent create new cloud event from cloud native events and pubsub
```
Comment (aneeshkp) [inline pkg/hwevent/event_ce.go:26]: space after //  for comments section

File: pkg/hwevent/event_ce_test.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
+// Copyright 2020 The Cloud Native Events Authors
+//
```
Comment (aneeshkp) [inline pkg/hwevent/event_ce_test.go:2]: rename event_ce_test.go to event_hw_test.go

File: pkg/hwevent/event_ce.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 The Cloud Native Events Authors
```
Comment (aneeshkp) [inline pkg/hwevent/event_ce.go:1]: rename event_ce.go to event_hw.go

File: pkg/hwevent/event_ce_test.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hwevent_test
+
+import (
+	"encoding/json"
```
Comment (aneeshkp) [inline pkg/hwevent/event_ce_test.go:18]: got format imports

File: v1/hwevent/event.go [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hwevent
+
+import (
+	"encoding/json"
```
Comment (aneeshkp) [inline v1/hwevent/event.go:18]: fmt go imports

File: v1/hwevent/event.go [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+	"github.com/redhat-cne/sdk-go/pkg/channel"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+
+	"github.com/redhat-cne/sdk-go/pkg/hwevent"
+)
+
+//PublishCloudEventToLog .. publish event data to a log
```
Comment (aneeshkp) [inline v1/hwevent/event.go:31]: space after //

File: v1/hwevent/event.go [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+func PublishCloudEventToLog(e cloudevents.Event) {
+	log.Infof("Publishing event to log %#v", e)
+}
+
+//CloudNativeEvent gets Cloud Native Event object
```
Comment (aneeshkp) [inline v1/hwevent/event.go:36]: add space after //

File: v1/hwevent/event.go [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+func CloudNativeEvent() hwevent.Event {
+	return hwevent.Event{Type: "HwEvent"}
```
Comment (aneeshkp) [inline v1/hwevent/event.go:38]: We should change this to use Enum eventually

**Comment by aneeshkp** [comment]
So you have created a new package and copied all files from the events folder, right? This is okay for testing the first iteration. But we should have a plan to merge these, or else we will accumulate technical debt. 
Please create an issue to track this.

**Comment by jzding** [comment]
> So you have created a new package and copied all files from the events folder, right? This is okay for testing the first iteration. But we should have a plan to merge these, or else we will accumulate technical debt.
> Please create an issue to track this.

Issue created https://github.com/redhat-cne/sdk-go/issues/24

File: v1/hwevent/event.go (hunk 9 above)
Comment (jzding) [reply v1/hwevent/event.go:38]: added TODO

File: pkg/hwevent/event_ce_test.go (hunk 5 above)
Comment (jzding) [reply pkg/hwevent/event_ce_test.go:18]: I run `go fmt ./...` and nothing changed?

Comment (aneeshkp) [reply pkg/hwevent/event_ce_test.go:18]: That won't do go imports formatting. right-click on your editor  and click on gotools -> goimports file

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**